opencensus/proto/agent/trace/v1/trace_service.proto

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+// NOTE: This proto is experimental and is subject to change at this point.
+// Please do not use it at the moment.
+
+package opencensus.proto.agent.trace.v1;
+
+import "opencensus/proto/agent/common/v1/common.proto";
+import "opencensus/proto/trace/trace.proto";
+
+option java_multiple_files = true;
+option java_package = "io.opencensus.proto.agent.trace.v1";
+option java_outer_classname = "TraceServiceProto";
+
+option go_package = "github.com/census-instrumentation/opencensus-proto/gen-go/agentproto/traceserviceproto/v1";
+
+service TraceService {
+
+  // After the initialization this RPC must be kept alive for the
+  // entire life of the application. Agent pushes configs to the
+  // application via a stream of responses.
+  rpc Config(stream ConfigTraceServiceRequest) returns (stream ConfigTraceServiceResponse) {}
+
+  // Allow to export Spans.
+  // For performance reason, it is recommended to keep this RPC
+  // alive for the entire life of the application.
+  rpc Export(stream ExportTraceServiceRequest) returns (stream ExportTraceServiceResponse) {}
+}
+
+message ConfigTraceServiceRequest {
+  // Identifier data effectively is a structured metadata. This is required only in the first message on the stream.
+  opencensus.proto.agent.common.v1.Node node = 1;
+
+  // Current configuration.
+  TraceConfig config = 2;
+}
+
+message ConfigTraceServiceResponse {
+  // Requested updated configuration.
+  TraceConfig config = 2;
+}
+
+message ExportTraceServiceRequest {
+  // Identifier data effectively is a structured metadata. This is required only in the first message on the stream.
+  opencensus.proto.agent.common.v1.Node node = 1;
+
+  repeated opencensus.proto.trace.Span spans = 2;
+}
+
+message ExportTraceServiceResponse {
+}
+
+// Global configuration of the trace service.
+message TraceConfig {
+
+  // The global default sampler used to make decisions on span sampling.
+  oneof sampler {
+    ProbabilitySampler probability_sampler = 1;
+
+    AlwaysSampler always_sampler = 2;
+
+    NeverSampler never_sampler = 3;
+
+    RateLimitingSampler rate_limiting_sampler = 4;
+  }
+
+  // TODO(songya): add more fields and move this to trace.proto.
```

Comment:
We should put it in a config.proto in the trace package. Can you do that?

Reply:
Done.